mslice/models/slice/mantid_slice_algorithm.py

```diff
@@ -32,14 +31,35 @@ def compute_slice(self, selected_workspace, x_axis, y_axis, smoothing, norm_to_o
         # perform number of events normalization then mask cells where no data was found
         with np.errstate(invalid='ignore'):
             plot_data = thisslice.getSignalArray() / thisslice.getNumEventsArray()
-        plot_data = np.ma.masked_where(np.isnan(plot_data), plot_data)
         # rot90 switches the x and y axis to to plot what user expected.
         plot_data = np.rot90(plot_data)
         self._workspace_provider.delete_workspace(thisslice)
         boundaries = [x_axis.start, x_axis.end, y_axis.start, y_axis.end]
         if norm_to_one:
             plot_data = self._norm_to_one(plot_data)
-        return plot_data, boundaries
+        plot = [None, None, None]
+        plot[0] = plot_data
+        plot[1] = self.compute_chi(plot_data, sample_temp, y_axis)
+        plot[2] = self.compute_chi_magnetic(plot[1])
+        return plot, boundaries
+
+    def compute_chi(self, scattering_data, sample_temp, y_axis):
+        if sample_temp is None:
+            return None
+        kBT = sample_temp * constants.value('Boltzmann constant in eV/K') * 1000  # convert to meV/K
```

Comment:
I would suggest extracting `constants.value('Boltzmann constant in eV/K') * 1000` out to a module-level variable to avoid `scipy` repeatedly looking up the value in its dictionary. Something like

```
KB_MEV = constants.value('Boltzmann constant in eV/K') * 1000
```

at the top of the module would suffice.